src/components/Dashboard/DashboardWidget.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { ReactNode } from "react";
+import {
+  Flex,
+  Card,
+  Text,
+  Box,
+  ThemePrepared,
+  SiteVariablesPrepared,
+  tabListBehavior,
+  Menu,
+} from "@fluentui/react-northstar";
+import { DashboardCallout, IWidgetActionKey } from "./DashboardCallout";
+
+export enum WidgetSize {
+  Single = "single",
+  Double = "double",
+  Triple = "triple",
+  Box = "box",
+}
+
+export interface IWidget {
+  size: WidgetSize;
+  title: string;
+  desc?: string;
+  widgetActionGroup?: IWidgetActionKey[];
+  body?: IWidgetBodyContent[];
+  link?: IWidgetLink;
+}
+
+export const Widget = ({
+  children,
+  size,
+}: {
+  children: ReactNode;
+  size: WidgetSize;
+}) => {
+  const cardStyle = {
+    gridColumnEnd: "auto",
+    gridRowEnd: "auto",
+    "@media (max-width: 842px)": {
+      gridColumnEnd: "span 3",
+    },
+  };
+  if (size === WidgetSize.Double) {
+    cardStyle.gridColumnEnd = "span 2";
+  }
+  if (size === WidgetSize.Box) {
+    cardStyle.gridColumnEnd = "span 2";
+    cardStyle.gridRowEnd = "span 2";
+  }
+  if (size === WidgetSize.Triple) {
+    cardStyle.gridColumnEnd = "span 3";
+  }
+  return (
+    <Card styles={cardStyle} fluid>
+      {children}
+    </Card>
+  );
+};
+
+export const WidgetTitle = ({
+  title,
+  desc,
+  globalTheme,
+  widgetActionGroup,
+}: {
+  title: string;
+  desc?: string;
+  globalTheme: ThemePrepared;
+  widgetActionGroup?: IWidgetActionKey[];
+}) => {
+  const [calloutOpen, setCalloutOpen] = React.useState(false);
+  return (
+    <Card.Header>
+      <Flex gap="gap.small" space="between" style={{ minHeight: "2rem" }}>
+        <Flex gap="gap.small" column>
+          <Text content={title} style={{ margin: 0 }} weight="bold" />
+          {desc && <Text content={desc} size="small" />}
+        </Flex>
+        <DashboardCallout
+          open={calloutOpen}
+          globalTheme={globalTheme}
+          onOpenChange={({ currentTarget }, props) => {
+            const open = !!props?.open;
+            setCalloutOpen(open);
+          }}
+          menuProps={{
+            offset: [0, 0],
+            position: "below",
+          }}
+          widgetActionGroup={widgetActionGroup}
+        />
+      </Flex>
+    </Card.Header>
+  );
+};
+
+const EmptyState = ({
+  siteVariables,
+}: {
+  siteVariables: SiteVariablesPrepared;
+}) => (
+  <Box
+    styles={{
+      height: "100%",
+      border: `1px dashed ${siteVariables}`,
```

Comment:
…how does this work? Isn't `SiteVariablesPrepared` a huge `Object`, not a CSS color?

Reply:
Thank you, color variable was missed